.gitignore

```diff
@@ -1,2 +1,6 @@
 target
 Cargo.lock
+
+# Vim
+.*.swp
+.*.swo
```

Comment:
Nit: ideally, we'd add this in a separate PR so it's not part of the diff for adding the mock `RwLock`.